src/player_event_handler.rs

```diff
@@ -14,9 +14,25 @@ pub struct EventHandler {
 impl EventHandler {
     pub fn new(mut player_events: PlayerEventChannel, onevent: &str) -> Self {
         let on_event = onevent.to_string();
-        let thread_handle = Some(thread::spawn(move || loop {
+
+        // The player increments the player id when it gets it...
+        let thread_name = format!(
+            "event-handler:{}",
+            PLAYER_COUNTER.load(Ordering::Relaxed).saturating_sub(1)
```

Comment:
Is Relaxed good enough here? No race conditions that could cause the builder to create two threads with the same name twice?

Reply:
The logic from player creation to when this is grabbed is synchronous so I don't see a problem but it doesn't hurt to make it stricter I guess. I can do that.

Reply:
I set all the ordering to Ordering::SeqCst there should never be a chance of a race condition.